CMakeLists.txt

```diff
@@ -8,7 +8,14 @@ list (APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 include(GNUInstallDirs)
 
 # choose the appropriate MSVC-runtime
-set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
+option(LIBCZI_DYNAMIC_MSVC_RUNTIME "In case of building with VisualStudio, use the dynamic C-runtime" OFF)
+
+# By default, the binaries built here will link to the static MSVC-runtime.
+# However, if the option LIBCZI_DYNAMIC_MSVC_RUNTIME is given, we will not instruct CMake to choose
+#  the static runtime, and it will default to the dynamic runtime (c.f. https://cmake.org/cmake/help/latest/variable/CMAKE_MSVC_RUNTIME_LIBRARY.html).
+if (NOT LIBCZI_DYNAMIC_MSVC_RUNTIME)
+ set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
+endif()
```

Comment:
I would suggest maybe the name of the flag should not have the word DYNAMIC in it - because if the library is being included by another upstream CMakeLists, then this will adopt the choice set in that upstream, whether it is the static or dynamic one.  it's more like "LIBCZI_DO_NOT_SET_MSVC_RUNTIME_LIBRARY"

Reply:
but aside from naming, this should definitely solve the problem of consistent linking with other codebases on Windows

Reply:
Yes, makes sense, I updated my PR accordingly. Thanks!